src/main/components/form/dateValidators.ts

```diff
@@ -0,0 +1,187 @@
+    return invalid;
+  }
+
+  for (const [fieldName, value] of Object.entries(date)) {
+    if (isNaN(+value)) {
+      return {
+        error: `${fieldName}NotANumber`,
+        fieldName,
+      };
+    }
+  }
+
+  const year = parseInt(date.year, 10) || 0;
+  const month = parseInt(date.month, 10) || 0;
+  const day = parseInt(date.day, 10) || 0;
+
+  if (day < 1 || day > 31) {
+    return {
+      error: 'dayInvalid',
+      fieldName: 'day',
+    };
+  }
+
+  if (month < 1 || month > 12) {
+    return {
+      error: 'monthInvalid',
+      fieldName: 'month',
+    };
+  }
+
+  if (year < 1000) {
+    return {
+      error: 'invalidYear',
+      fieldName: 'year',
+    };
+  }
+
+  const enteredDate = new Date(+date.year, +date.month - 1, +date.day);
+  const dateMinus100 = new Date();
+  dateMinus100.setFullYear(dateMinus100.getFullYear() - 100);
+  dateMinus100.setHours(0, 0, 0, 0);
+
+  if (enteredDate < dateMinus100) {
+    return { error: 'invalidDateTooFarInPast', fieldName: 'year' };
+  }
+
+  if (validateDayInTheMonth(date)) {
+    return { error: invalid, fieldName: validateDayInTheMonth(date) as string };
+  }
+};
+
+export const validateDayInTheMonth = (date: CaseDate): string | boolean => {
+  const year = parseInt(date.year, 10) || 0;
+
+  const month = parseInt(date.month, 10) || 0;
+  const day = parseInt(date.day, 10) || 0;
+
+  const jsDate = dayjs(new Date(year, month - 1, day));
+
+  const yearValid = jsDate.year() === year;
+  const monthValid = jsDate.month() + 1 === month;
+  const dayValid = jsDate.date() === day;
+
+  return !yearValid ? 'year' : !monthValid ? 'month' : !dayValid ? 'day' : false;
+};
+
+export const areDates10YearsApartOrMore: DateValidator = (
+  caseDate1: CaseDate | undefined,
+  caseDate2: CaseDate | undefined
+) => {
+  const date1 = new Date(+caseDate1.year, +caseDate1.month - 1, +caseDate1.day);
+  const date2 = new Date(+caseDate2.year, +caseDate2.month - 1, +caseDate2.day);
+  const yearsBetweenDates = new Date(Math.abs(date1.getTime() - date2.getTime())).getUTCFullYear() - 1970;
+
+  if (yearsBetweenDates < 10) {
+    return { error: 'invalidDateTooRecent', fieldName: 'day' };
+  }
+};
+
+export const isDateTenYearsInPast: DateValidator = (date: CaseDate | undefined) => {
+  const enteredDate = new Date(+date.year, +date.month - 1, +date.day);
+  const dateMinus10 = new Date();
+  dateMinus10.setFullYear(dateMinus10.getFullYear() - 10);
+  dateMinus10.setHours(0, 0, 0, 0);
+
+  if (enteredDate < dateMinus10) {
+    return { error: 'invalidDateMoreThanTenYearsInPast', fieldName: 'year' };
+  }
+};
+
+export const isDateTenYearsInFuture: DateValidator = (date: CaseDate | undefined) => {
+  const enteredDate = new Date(+date.year, +date.month - 1, +date.day);
+  const datePlus10 = new Date();
+  datePlus10.setFullYear(datePlus10.getFullYear() + 10);
+
+  if (enteredDate > datePlus10) {
+    return { error: 'invalidDateMoreThanTenYearsInFuture', fieldName: 'year' };
+  }
+};
+
+/**
```

Comment:
Not sure what our rules are for JSDocs but why do some methods have them and others not?

Reply:
I see that not many places have them, I'll try and make the method names a lot clearer.